LoopKitHostedTests/CoreDataMigrationTests.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  CoreDataMigrationTests.swift
+//  LoopKitHostedTests
+//
+//  Created by Rick Pasetto on 8/9/21.
+//  Copyright © 2021 LoopKit Authors. All rights reserved.
+//
+
+import CoreData
+import HealthKit
+import XCTest
+@testable import LoopKit
+
+class CoreDataV1MigrationTests: XCTestCase {
+    
+    var cacheStore: PersistenceController!
+    let fileManager = FileManager()
+    static let bundleURL = Bundle(for: CoreDataV1MigrationTests.self).bundleURL
+
+    override func setUpWithError() throws {
+        try? fileManager.removeItem(at: Self.bundleURL.appendingPathComponent("Model.sqlite"))
+        try fileManager.copyItem(at: Self.bundleURL.appendingPathComponent("Model.sqlite.v1.original"),
+                                 to: Self.bundleURL.appendingPathComponent("Model.sqlite"))
+    }
+
+    override func tearDownWithError() throws {
+        
+        // remove all stores
+        try cacheStore.managedObjectContext.persistentStoreCoordinator?.persistentStores.forEach { store in
+            try cacheStore.managedObjectContext.persistentStoreCoordinator?.remove(store)
+        }
+        
+        try fileManager.removeItem(at: Self.bundleURL.appendingPathComponent("Model.sqlite"))
+        cacheStore = nil
+    }
+
+    func testV1Migration() throws {
+        let e = expectation(description: #function)
+        cacheStore = PersistenceController.init(directoryURL: Self.bundleURL)
+        var error: Error?
+        cacheStore.onReady {
+            if let err = $0 {
+                error = err
+                XCTFail("Error opening \(Self.bundleURL): \(err)")
+            }
+            e.fulfill()
+        }
+        wait(for: [e], timeout: 10.0)
+        if let error = error { throw error }
+//        print("\(String(describing: cacheStore.managedObjectContext.persistentStoreCoordinator?.managedObjectModel.entities))")
+        
+        var entries: [CachedCarbObject]!
+        let e0 = expectation(description: #function)
```

Comment:
Using `#function` is clever. In this case, since both expectations in the same function are using it, a failed expectation will be ambiguous, so might not want to do that.

Reply:
copy/paste error on my part, will fix 👍